src/main/java/fi/oph/akt/api/dto/clerk/modify/TranslatorCreateDTO.java

```diff
@@ -3,12 +3,12 @@
 import java.util.List;
 import javax.validation.constraints.NotBlank;
 import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
 import lombok.Builder;
-import lombok.NonNull;
 
 public record TranslatorCreateDTO(
-  @NonNull @NotBlank String firstName,
-  @NonNull @NotBlank String lastName,
+  @NotBlank String firstName,
```

Comment:
Lombokin NonNull oli sen vuoksi ettei koodissa voi luoda vääränlaista dto:ta. Ehkä niitä ei tarvita, täytyyhän tuossa vähän aikaa miettiä miksi on lombokin NonNull ja validation NotBlank, ehkä selkeämpää näin lukea, mutta onko selkeämpää käyttää?

Reply:
Tosin, jos nyt otetaan nuo lombokin NonNull pois, niin sitten otetaan pois muistakin dto:sta.

Reply:
Vähän tätä aiemmin ihmettelinkin, mutta lombok.NonNull ja javax.validation.NotNull välinen ero ei ollut lainkaan selvä. Ilmeisesti molempien tapauksessa (merkattiin tietty ei-nullable kenttä NonNullina tai NotNullina) clientille päätyy kuitenkin jonkinmoinen virhe esimerkiksi siitä, että kentän arvo puuttui, jos pyynnössä ei tuota kenttää ollut.

Mutta tuota puolta ei tullut niinkään ajateltua, että NonNull määreellä voidaan koodin tasolla estää olioiden luonti null-arvoilla ko. kenttien osalta. Mun puolesta ok palauttaakin NonNull määreet näihin paikkoihin, joissa vast. javax validaatio (NotNull, NotEmpty tai NotBlank).

Reply:
Juu, ei ne vahingossa sinne ole tulleet. Lombokin annotointi estää ainakin koodissa (ellei jopa http rajapinnasta tulevaa) luomasta null arvolla, kun taas validation annotointi dokumentoi näkyy swagger dokumentaatiossa ja toimii kun http rajapinnasta dto luodaan.